File: frontend/src/api/resourcePlanLabor.ts
import request from '@/utils/request'
import type { R } from '@/types/api'

export interface ResourcePlanLabor {
  id?: number
  projectId: number
  wbsCode: string
  wbsName?: string
  workTypeCode: string
  workTypeName?: string
  laborCategoryCode?: string
  laborCategoryName?: string
  planQuantity?: number
  planStartDate?: string
  planEndDate?: string
  actualQuantity?: number
  actualStartDate?: string
  actualEndDate?: string
  status?: string
  approvalStatus?: string
  warningStatus?: string
  remark?: string
  deleted?: number
}

export interface ProgressDTO {
  actualQuantity?: number
  actualStartDate?: string
  actualEndDate?: string
  remark?: string
  attendanceRecords?: string
}

export interface LaborChangeDTO {
  planId?: number
  changeType?: string
  changeReason?: string
  changeDetails?: string
  newPlanStartDate?: string
  newPlanEndDate?: string
  newLaborCategoryCode?: string
  newLaborCategoryName?: string
}

export interface WarningStatusVO {
  warningLevel?: string
  warningMessage?: string
  suggestion?: string
  checkTime?: string
}

/**
 * 查询劳动力需求列表
 */
export function getLaborList(params: { 
  projectId?: number, 
  status?: string,
  warningStatus?: string,
  workTypeCode?: string,
  laborCategoryCode?: string 
}): Promise<R<ResourcePlanLabor[]>> {
  return request({
    url: '/v1/resource-plan/labor/list',
    method: 'get',
    params
  })
}

/**
 * 根据ID查询劳动力需求
 */
export function getLaborById(id: number): Promise<R<ResourcePlanLabor>> {
  return request({
    url: `/v1/resource-plan/labor/${id}`,
    method: 'get'
  })
}

/**
 * 新增劳动力需求
 */
export function createLabor(data: ResourcePlanLabor): Promise<R<boolean>> {
  return request({
    url: '/v1/resource-plan/labor',
    method: 'post',
    data
  })
}

/**
 * 修改劳动力需求
 */
export function updateLabor(id: number, data: ResourcePlanLabor): Promise<R<boolean>> {
  return request({
    url: `/v1/resource-plan/labor/${id}`,
    method: 'put',
    data
  })
}

/**
 * 删除劳动力需求
 */
export function deleteLabor(id: number): Promise<R<boolean>> {
  return request({
    url: `/v1/resource-plan/labor/${id}`,
    method: 'delete'
  })
}

/**
 * 提交审批
 */
export function submitLabor(id: number): Promise<R<boolean>> {
  return request({
    url: `/v1/resource-plan/labor/${id}/submit`,
    method: 'post'
  })
}

/**
 * 登记进展
 */
export function registerProgress(id: number, data: ProgressDTO): Promise<R<boolean>> {
  return request({
    url: `/v1/resource-plan/labor/${id}/progress`,
    method: 'post',
    data
  })
}

/**
 * 查询预警状态
 */
export function getWarningStatus(id: number): Promise<R<WarningStatusVO>> {
  return request({
    url: `/v1/resource-plan/labor/${id}/warning-status`,
    method: 'get'
  })
}

/**
 * 查询变更历史
 */
export function getChangeHistory(id: number): Promise<R<any[]>> {
  return request({
    url: `/v1/resource-plan/labor/${id}/change-history`,
    method: 'get'
  })
}

/**
 * 发起变更申请
 */
export function createChange(data: LaborChangeDTO): Promise<R<number>> {
  return request({
    url: '/v1/resource-plan/labor/change',
    method: 'post',
    data
  })
}

File: backend/src/main/java/com/company/rpw/entity/ResourcePlanLabor.java
package com.company.rpw.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.company.rpw.common.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

/**
 * 劳动力计划实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("resource_plan_labor")
public class ResourcePlanLabor extends BaseEntity {

    /** 项目ID */
    private Long projectId;

    /** WBS编码 */
    private String wbsCode;

    /** WBS名称（冗余存储，减少关联查询） */
    private String wbsName;

    /** 工种编码 */
    private String workTypeCode;

    /** 工种名称（冗余存储） */
    private String workTypeName;

    /** 劳务类别编码 */
    private String laborCategoryCode;

    /** 劳务类别名称（冗余存储） */
    private String laborCategoryName;

    /** 计划人数 */
    private Integer planQuantity;

    /** 计划开始日期 */
    private LocalDate planStartDate;

    /** 计划结束日期 */
    private LocalDate planEndDate;

    /** 实际人数 */
    private Integer actualQuantity;

    /** 实际开始日期 */
    private LocalDate actualStartDate;

    /** 实际结束日期 */
    private LocalDate actualEndDate;

    /** 状态（DRAFT-草稿/SUBMITTED-已提交/IN_PROGRESS-进行中/COMPLETED-已完成/TERMINATED-已终止） */
    private String status;

    /** 审批状态（DRAFT-草稿/SUBMITTED-已提交/APPROVED-已批准/REJECTED-已驳回） */
    private String approvalStatus;

    /** Flowable流程实例ID */
    private String processInstanceId;

    /** 出勤记录（JSON格式） */
    private String attendanceRecords;

    /** 备注 */
    private String remark;
}
